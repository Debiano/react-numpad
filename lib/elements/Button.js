import styled from 'styled-components';

const Button = styled.button`
  background: #fff;    
  border: none;
  color: #3a3a3a;
  cursor: pointer;
  display: inline-block;
  font-size: 16px;
  line-height: 40px;
  font-weight: 200;
  margin: 3px 0;
  outline: none;
  padding: 0 12px;
  border-radius: 2px;
  text-transform: uppercase;
  transition:
    transform 0.08s ease,
    background-color 0.08s ease,
    box-shadow 0.08s ease;

  &:active {
    transform: translateY(1.5px);
    box-shadow: 
        inset 0 -0.5px 0 #aaaaaa,
    inset 0px -1px 1px -1px #fff,
        0px 0.5px 0.75px 0px #999;
  }
`;

export default Button;
