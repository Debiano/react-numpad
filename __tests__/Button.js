import React from 'react'
import Button from '../lib/elements/Button'
import renderer from 'react-test-renderer'

test('Button console log when clicked', () => {
  const component = renderer.create(
    <Button className='NumPad-keypad-button' onClick={() => console.log('ciao')}>Ciao</Button>
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});