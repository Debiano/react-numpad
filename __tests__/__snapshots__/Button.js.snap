// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button console log when clicked 1`] = `
<button
  className="NumPad-keypad-button sc-bdVaJa gHittV"
  onClick={[Function]}
>
  Ciao
</button>
`;

exports[`Button console log when clicked 2`] = `
<button
  className="NumPad-keypad-button sc-bdVaJa gHittV"
  onClick={[Function]}
>
  Ciao
</button>
`;
